FrameRate/FrameRateImplementation.cpp

```diff
@@ -0,0 +1,509 @@
+#include "UtilsIarm.h"
+#include "UtilsProcess.h"
+
+//Defines
+#define DEFAULT_FPS_COLLECTION_TIME_IN_MILLISECONDS 10000
+#define MINIMUM_FPS_COLLECTION_TIME_IN_MILLISECONDS 100
+#define DEFAULT_MIN_FPS_VALUE 60
+#define DEFAULT_MAX_FPS_VALUE -1
+
+namespace WPEFramework
+{
+    namespace Plugin
+    {
+        SERVICE_REGISTRATION(FrameRateImplementation, 1, 0);
+        FrameRateImplementation* FrameRateImplementation::_instance = nullptr;
+
+        FrameRateImplementation::FrameRateImplementation()
+        {
+            FrameRateImplementation::_instance = this;
+            InitializeIARM();
+        }
+
+        FrameRateImplementation::~FrameRateImplementation()
+        {
+            DeinitializeIARM();
+        }
+
+        // IARM EventHandler
+        static void _frameEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len);
+
+        void FrameRateImplementation::InitializeIARM()
+        {
+            TRACE(Trace::Information, (_T("FrameRateImplementation::InitializeIARM()")));
+            if (Utils::IARM::init())
+            {
+                IARM_Result_t res;
+                IARM_CHECK(IARM_Bus_RegisterEventHandler(IARM_BUS_DSMGR_NAME, IARM_BUS_DSMGR_EVENT_DISPLAY_FRAMRATE_PRECHANGE, _frameEventHandler));
+                IARM_CHECK(IARM_Bus_RegisterEventHandler(IARM_BUS_DSMGR_NAME, IARM_BUS_DSMGR_EVENT_DISPLAY_FRAMRATE_POSTCHANGE, _frameEventHandler));
+            }
+        }
+
+        void FrameRateImplementation::DeinitializeIARM()
+        {
+			TRACE(Trace::Information, (_T("FrameRateImplementation::DeinitializeIARM()")));
+            if (Utils::IARM::isConnected())
+            {
+                IARM_Result_t res;
+                IARM_CHECK(IARM_Bus_RemoveEventHandler(IARM_BUS_DSMGR_NAME, IARM_BUS_DSMGR_EVENT_DISPLAY_FRAMRATE_PRECHANGE, _frameEventHandler));
+                IARM_CHECK(IARM_Bus_RemoveEventHandler(IARM_BUS_DSMGR_NAME, IARM_BUS_DSMGR_EVENT_DISPLAY_FRAMRATE_POSTCHANGE, _frameEventHandler));
+            }
+        }
+
+        void FrameRateImplementation::dispatchOnDisplayFrameRateChangingEvent(const string& displayFrameRate)
+        {
+            std::list<Exchange::IFrameRate::INotification*>::const_iterator index(_framerateNotification.begin());
+            while (index != _framerateNotification.end())
+            {
+                (*index)->OnDisplayFrameRateChanging(displayFrameRate);
+                ++index;
+            }
+        }
+
+        void FrameRateImplementation::dispatchOnDisplayFrameRateChangedEvent(const string& displayFrameRate)
+        {
+            std::list<Exchange::IFrameRate::INotification*>::const_iterator index(_framerateNotification.begin());
+            while (index != _framerateNotification.end())
+            {
+                (*index)->OnDisplayFrameRateChanged(displayFrameRate);
+                ++index;
+            }
+        }
+
+        void FrameRateImplementation::Dispatch(Event event, const JsonValue params)
+        {
```

Comment:
Even though this logic works, please make this function similar to other plugin's Dispatch, dispatchEvent to keep the interfaces uniform across all plugins.